### ISSUE TITLE
`Kirby\Cms\Field` shouldn't take optional parameter as first parameter of constructor

### ISSUE THREAD

**Issue by distantnative** [getkirby/kirby#3262]
- `$parent` should go last and be optional
- It would also simplify a couple of our use cases if key and value are just set as the same if only one parameter is passed

https://github.com/getkirby/kirby/blob/master/src/Cms/Field.php#L99

**Comment by lukasbestle** [comment]
I agree `$parent` should be the last param in the long run. How can we best migrate to this without breaking-changes in plugins etc. that directly use the constructor?

> It would also simplify a couple of our use cases if key and value are just set as the same if only one parameter is passed

In which places do we have this use-case at the moment?

**Comment by distantnative** [comment]
I haven't had a good idea how this wouldn't be a breaking change.

I saw this behavior working with the Nest classes.

**Comment by afbora** [comment]
Just an idea: can it be solved with `func_get_args()`? Then in 3.7, the parameters are stabilized 🤔

**Comment by lukasbestle** [comment]
Maybe we don't even need `func_get_args()`:

We could reorder the params and remove the strict type hints for now. Then we could check in the method if the first param (now named `$key`) is a `Model` object. If it is, we assume the old behavior and reassign the variables to the old order. If it isn't, we assume the new behavior and keep the values as provided.

In 3.7 we could then remove the `Model` check and add the strict type hints back in.

**Comment by distantnative** [comment]
> In 3.7 we could then remove the Model check and add the strict type hints back in.
 
 Will this make a big difference to doing this in 3.6 already?
 The breaking change will cause work for the few developers using this (I guess including us) anyways.
 
 Ok of course if we add `deprecated()` warnings, might help to ease the transition.

**Comment by lukasbestle** [comment]
> Ok of course if we add `deprecated()` warnings, might help to ease the transition.

Yep, that's what I was thinking.

**Comment by distantnative** [comment]
Given the breaking changes in 3.6.0 already, I would postpone this one to 3.7 or 3.8 - I don't think this one here is urgent.
